tests/pytorch/test_pytorch_autolog.py

```diff
@@ -0,0 +1,158 @@
+import os
+import pytest
+import pytorch_lightning as pl
+import torch
+from iris import IrisClassification
+import mlflow
+import mlflow.pytorch
+from pytorch_lightning.callbacks.early_stopping import EarlyStopping
+from pytorch_lightning.callbacks import ModelCheckpoint
+
+NUM_EPOCHS = 20
+
+
+@pytest.fixture
+def pytorch_model():
+    mlflow.pytorch.autolog()
+    model = IrisClassification()
+    trainer = pl.Trainer(max_epochs=NUM_EPOCHS)
+    trainer.fit(model)
+    client = mlflow.tracking.MlflowClient()
+    run = client.get_run(client.list_run_infos(experiment_id="0")[0].run_id)
+    return trainer, run
+
+
+def test_pytorch_autolog_logs_default_params(pytorch_model):
+    _, run = pytorch_model
+    data = run.data
+    assert "learning_rate" in data.params
+    assert "epsilon" in data.params
+    assert "optimizer_name" in data.params
+    assert "weight_decay" in data.params
+    assert "betas" in data.params
+
+
+def test_pytorch_autolog_logs_expected_data(pytorch_model):
+    _, run = pytorch_model
+    data = run.data
+
+    # Checking if metrics are logged
+    assert "loss" in data.metrics
+    assert "val_loss" in data.metrics
+
+    # Testing unwanted parameters are not logged
+    assert "callbacks" not in data.params
+
+    # Testing optimizer parameters are logged
+    assert "optimizer_name" in data.params
+    assert data.params["optimizer_name"] == "Adam"
+
+    # Testing model_summary.txt is saved
+    client = mlflow.tracking.MlflowClient()
+    artifacts = client.list_artifacts(run.info.run_id)
+    artifacts = map(lambda x: x.path, artifacts)
+    assert "model_summary.txt" in artifacts
+
+
+# pylint: disable=unused-argument
+def test_pytorch_autolog_persists_manually_created_run(pytorch_model):
+    with mlflow.start_run() as manual_run:
+        assert mlflow.active_run() is not None
+        assert mlflow.active_run().info.run_id == manual_run.info.run_id
```

Comment:
This test should look like:

```python
    mlflow.pytorch.autolog()

    with mlflow.start_run() as run:
        # insert code to train a model

        assert mlflow.active_run()
        assert mlflow.active_run().info.run_id == run.info.
```

Reply:
makes sense. Ideally, the run should start first and then the autolog method should be called. will change it.

Reply:
The intention of the test case is to check if autolog persiste manually created run. I feel the code should be like this 

```

    with mlflow.start_run() as run:
        mlflow.pytorch.autolog()
        # insert code to train a model

        assert mlflow.active_run()
        assert mlflow.active_run().info.run_id == run.info.run_id

```

User creates a run and then calls autolog method. Autolog should check for active run and since the run is already active, it should persist the same run (should  not create a new one). Please correct me if my understanding is wrong.